Data/Functor/Foldable.hs

```diff
@@ -349,19 +346,31 @@ instance Functor f => Recursive (Free f a) where
   project (Pure a) = CMTF.Pure a
   project (Free f) = CMTF.Free f
 
+improveF :: Functor f => CMFC.F f a -> Free f a
+improveF x = CMFC.improve (CMFC.fromF x)
+-- | It may be better to work with the instance for `CMFC.F` directly.
 instance Functor f => Corecursive (Free f a) where
   embed (CMTF.Pure a) = Pure a
   embed (CMTF.Free f) = Free f
+  ana               coalg = improveF . ana               coalg
+  postpro       nat coalg = improveF . postpro       nat coalg
+  gpostpro dist nat coalg = improveF . gpostpro dist nat coalg
 
 -- | Free transformations of monads are Recursive/Corecursive
 type instance Base (FreeT f m a) = Compose m (FreeF f a)
-
 instance (Functor m, Functor f) => Recursive (FreeT f m a) where
   project = Compose . runFreeT
-
 instance (Functor m, Functor f) => Corecursive (FreeT f m a) where
   embed = FreeT . getCompose
 
+-- If you are looking for instances for the free MonadPlus, please use the
```

Comment:
Well, it is visible in the source code. ;)

I'm not sure where it would be best to put it for haddock.